src/physical_exec.rs

```diff
@@ -152,12 +152,21 @@ impl FileOpener for OrcOpener {
         let reader =
             ObjectStoreReader::new(self.object_store.clone(), file_meta.object_meta.clone());
         let batch_size = self.batch_size;
-        // Offset by 1 since index 0 is the root
-        let projection = self.projection.iter().map(|i| i + 1).collect::<Vec<_>>();
+        let projected_schema = SchemaRef::from(self.table_schema.project(&self.projection)?);
+
         Ok(Box::pin(async move {
             let mut builder = ArrowReaderBuilder::try_new_async(reader)
                 .await
                 .map_err(ArrowError::from)?;
+            // Find complex data type column index as projection
+            let mut projection = Vec::with_capacity(projected_schema.fields().len());
+            for named_column in builder.file_metadata().root_data_type().children() {
+                if let Some((_table_idx, _table_field)) =
+                    projected_schema.fields().find(named_column.name())
+                {
+                    projection.push(named_column.data_type().column_index());
+                }
+            }
```

Comment:
```suggestion
            let projection = builder
                .file_metadata()
                .root_data_type()
                .children()
                .iter()
                .filter(|named_column| {
                    projected_schema
                        .fields()
                        .find(named_column.name())
                        .is_some()
                })
                .map(|named_column| named_column.data_type().column_index());
```

Thoughts on using a more iterator based approach? Could potentially be simplified further.